homeassistant/components/twentemilieu/sensor.py

```diff
@@ -37,118 +64,46 @@ async def async_setup_entry(
-        TwenteMilieuSensor(
-            twentemilieu,
-            unique_id=entry.data[CONF_ID],
-            name=f"{WASTE_TYPE_ORGANIC} Waste Pickup",
-            waste_type=WASTE_TYPE_ORGANIC,
-            icon="mdi:delete-empty",
-        ),
-        TwenteMilieuSensor(
-            twentemilieu,
-            unique_id=entry.data[CONF_ID],
-            name=f"{WASTE_TYPE_PAPER} Waste Pickup",
-            waste_type=WASTE_TYPE_PAPER,
-            icon="mdi:delete-empty",
-        ),
-        TwenteMilieuSensor(
-            twentemilieu,
-            unique_id=entry.data[CONF_ID],
-            name=f"{WASTE_TYPE_PLASTIC} Waste Pickup",
-            waste_type=WASTE_TYPE_PLASTIC,
-            icon="mdi:delete-empty",
-        ),
-    ]
-
-    async_add_entities(sensors, True)
+    async_add_entities(
+        [
+            TwenteMilieuSensor(twentemilieu, entry.data[CONF_ID], description)
+            for description in SENSORS
+        ],
+        True,
+    )
 
 
 class TwenteMilieuSensor(SensorEntity):
     """Defines a Twente Milieu sensor."""
 
-    _attr_device_class = DEVICE_CLASS_DATE
+    _attr_should_poll = False
 
     def __init__(
         self,
         twentemilieu: TwenteMilieu,
         unique_id: str,
-        name: str,
-        waste_type: str,
-        icon: str,
+        description: SensorEntityDescription,
     ) -> None:
         """Initialize the Twente Milieu entity."""
-        self._available = True
-        self._unique_id = unique_id
-        self._icon = icon
-        self._name = name
+        self.entity_description = description
         self._twentemilieu = twentemilieu
-        self._waste_type = waste_type
-
-        self._state = None
-
-    @property
-    def name(self) -> str:
-        """Return the name of the entity."""
-        return self._name
-
-    @property
-    def icon(self) -> str:
-        """Return the mdi icon of the entity."""
-        return self._icon
-
-    @property
-    def available(self) -> bool:
-        """Return True if entity is available."""
-        return self._available
-
-    @property
-    def unique_id(self) -> str:
-        """Return the unique ID for this sensor."""
-        return f"{DOMAIN}_{self._unique_id}_{self._waste_type}"
-
-    @property
-    def should_poll(self) -> bool:
-        """Return the polling requirement of the entity."""
-        return False
+        self._attr_unique_id = f"{DOMAIN}_{unique_id}_{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            manufacturer="Twente Milieu",
+            name="Twente Milieu",
+        )
 
     async def async_added_to_hass(self) -> None:
         """Connect to dispatcher listening for entity data notifications."""
         self.async_on_remove(
             async_dispatcher_connect(
-                self.hass, DATA_UPDATE, self._schedule_immediate_update
+                self.hass, DATA_UPDATE, self.async_schedule_update_ha_state
```

Comment:
There will be no call to `async_update` now when the callback is called as it was done before. It looks like the state will never update again now after the first update.